h2o-core/src/main/java/water/fvec/RollupStats.java

```diff
@@ -414,6 +418,8 @@ public void compute2() {
         if (rs != null) {
           if (rs.isReady()) {
             if (_computeHisto && !rs.hasHisto()) { // a.2 => compute rollups
+              final Vec vec = DKV.getGet(_vecKey);
```

Comment:
This is a bit subtle but I think it's better not to check here and let it run and fail instead (either NPE cause data is gone already or when installing the response). Otherwise I think it can return rollups with no error and no histo and cause an infinite loop on the caller's site (note that in Keyed.remove, the object itself is removed from DKV before remove_impl is called).

I think following can happen:

a) thread 1 requests rollups with histo, reaches home node, sees there is RS with no histo
b) thread 2 removes the Vec, start processing remove_impl
c) thread 1 sees there is no vec, returns, however, this rollups has not yet been marked as mutating by 2 so all is good and rollups comes back with no histo.